src/spatial/nsm.jl

```diff
@@ -110,18 +108,19 @@ end
 reevaluate all rates, recalculate tentative site firing times, and reinit the priority queue
 """
 function fill_rates_and_get_times!(aggregation::NSMJumpAggregation, u, t)
-    @unpack ma_jumps, hopping_constants, spatial_system, cur_rates = aggregation
+    @unpack spatial_system, rx_rates, hop_rates, rng, numspecies = aggregation
 
-    reset!(cur_rates)
-    num_majumps = get_num_majumps(ma_jumps)
-    num_species = size(u,1) #NOTE assumes u is a matrix with ith column being the ith site
-    num_sites = DiffEqJump.num_sites(spatial_system)
+    reset!(rx_rates)
+    reset!(hop_rates)
 
-    pqdata = Vector{typeof(t)}(undef, num_sites)
-    for site in 1:num_sites
-        update_reaction_rates!(cur_rates, 1:num_majumps, u, ma_jumps, site)
-        update_hop_rates!(cur_rates, 1:num_species, hopping_constants, u, site, spatial_system)
-        pqdata[site] = t + randexp(aggregation.rng) / total_site_rate(cur_rates, site)
+    num_reactions = num_rxs(rx_rates)
+    num_nodes = num_sites(spatial_system)
+
+    pqdata = Vector{typeof(t)}(undef, num_nodes)
```

Comment:
Should this be reset too?

Reply:
What do you mean? `pqdata` is not stored in the aggregation, so it cannot be reset.

Reply:
I think it is stored in the priority queue though, so you might be able to reuse it. We can save that for the next PR.